### PR TITLE
Frontend button and toggle styling

### PR DESCRIPTION
**Pull request by thisconnect**
The background color of unchecked toggles were hard to see in dark
mode, background tertiary is also used for secondary buttons.

Changed background tertiary in darkmode to fix the toggle color,
this also changes button secondary color.

Another issue is that button secondary and button transparent are
visually very similar, but the transparent style is not a text only
transparent button as the name implies. This can be improved in a
subsequent change.

Also added stricter types for button and a cleanup commit that removes
duplicate CSS properties.

### CODE REVIEW

File: frontends/web/src/components/forms/button.module.css [hunk 1]
```diff
@@ -1,27 +1,25 @@
...
     border-color: transparent;
     border-radius: 2px;
+    border-style: solid;
+    border-width: 1px;
     cursor: default;
     /* Otter doesn't like display: inline-flex; */
     display: inline-flex;
     flex-direction: row;
-    justify-content: center;
-    align-items: center;
```
Comment (thisconnect) [inline frontends/web/src/components/forms/button.module.css:12]: removed duplicate styles

File: frontends/web/src/style/variables.css [hunk 2]
```diff
@@ -170,7 +170,7 @@
         --background-secondary: var(--color-softblack);
         --background-dark: var(--color-softblack);
         --background-focus: var(--color-softblack);
-        --background-tertiary: var(--color-dark);
+        --background-tertiary: var(--color-gray);
```
Comment (thisconnect) [inline frontends/web/src/style/variables.css:173]: `--background-tertiary` is only used for toggles and secondary buttons. This changes darkmode color of unchecked toggle background and button secondary border. I'll paste screenshots in the comments.

https://github.com/search?q=repo%3Adigitalbitbox%2Fbitbox-wallet-app%20--background-tertiary&type=code

File: frontends/web/src/components/forms/button.module.css [hunk 3]
```diff
@@ -1,27 +1,25 @@
...
     font-family: var(--font-family);
     font-size: var(--size-default);
     font-weight: 400;
     justify-content: center;
     min-width: 100px !important;
     height: 50px;
+    outline: none;
     padding: 0 var(--space-default);
     position: relative;
-    outline: none;
```
Comment (thisconnect) [inline frontends/web/src/components/forms/button.module.css:21]: sorted alphabetically, so it is easy to detect duplicates (this one was fine)

File: frontends/web/src/components/forms/button.module.css [hunk 4]
```diff
@@ -1,27 +1,25 @@
...
     text-align: center;
     text-decoration: none;
-    -webkit-appearance: none;
```
Comment (thisconnect) [inline frontends/web/src/components/forms/button.module.css:24]: unprefixed appearance property.

**Comment by thisconnect** [comment]
Screen Shot 2023-07-13 at 14 01 55

(4.38.0)

Screen Shot 2023-07-13 at 14 01 40

(with this change)

**Comment by thisconnect** [comment]
Screen Shot 2023-07-12 at 17 57 32

Screen Shot 2023-07-12 at 17 57 26

**Comment by thisconnect** [comment]
Screen Shot 2023-07-13 at 09 43 31

**Comment by thisconnect** [comment]
Screen Shot 2023-07-13 at 09 44 43

**Comment by thisconnect** [comment]
webdev

    
Screen Shot 2023-07-13 at 11 32 27

File: frontends/web/src/components/forms/button.tsx [hunk 5]
```diff
@@ -19,25 +19,34 @@ import { ComponentPropsWithoutRef, ReactNode } from 'react';
...
+  | ({ primary: true } & Omit<TButtonStyleBase, 'primary'>)
+  | ({ secondary: true } & Omit<TButtonStyleBase, 'secondary'>)
+  | ({ transparent: true } & Omit<TButtonStyleBase, 'transparent'>)
+
+type TButtonStyleBase = {
+  danger?: false;
+  primary?: false;
+  secondary?: false;
+  transparent?: false;
+}
```
Comment (shonsirsha) [inline frontends/web/src/components/forms/button.tsx:33]: Nice! 👍

**Review by shonsirsha — APPROVED** [review]
🚀  w/ a small nit (feel free to ignore though).

File: frontends/web/src/routes/device/bitbox01/restore.tsx [hunk 6]
```diff
@@ -155,13 +163,21 @@ class Restore extends Component<Props, State> {
...
+                      {t('button.restore')}
+                    </Button>
+                  ) : (
+                    <Button
+                      type="submit"
+                      primary
+                      disabled={!understand || !this.validate() || isConfirming}>
+                      {t('button.restore')}
+                    </Button>
+                  ) }
```
[the remark this answers is not in the document]
Comment (thisconnect) [reply frontends/web/src/routes/device/bitbox01/restore.tsx:180]: much nicer thanks 👍 

I'll update the PR.

[the remark this answers is not in the document]
Comment (thisconnect) [reply frontends/web/src/routes/device/bitbox01/restore.tsx]: done

**Pull request merged**